.github/workflows/wheels.yml

```diff
@@ -0,0 +1,80 @@
+name: cuDF wheels
+
+on:
+  push:
+    branches:
+      - 'pull-request/[0-9]+'
+  workflow_call:
+    inputs:
+      versioneer-override:
+        type: string
+        default: ''
+      build-tag:
+        type: string
+        default: ''
+      branch:
+        required: true
+        type: string
+      date:
+        required: true
+        type: string
+      sha:
+        required: true
+        type: string
+      build-type:
+        type: string
+        default: nightly
+
+concurrency:
+  group: "cudf-${{ github.workflow }}-${{ github.ref }}"
+  cancel-in-progress: true
+
+jobs:
+  cudf-wheels:
+    uses: rapidsai/shared-action-workflows/.github/workflows/wheels-manylinux.yml@feat/pure-python-wheels
+    with:
+      repo: rapidsai/cudf
+
+      build-type: ${{ inputs.build-type || 'pull-request' }}
+      branch: ${{ inputs.branch }}
+      sha: ${{ inputs.sha }}
+      date: ${{ inputs.date }}
+
+      package-dir: python/cudf
+      package-name: cudf
+
+      python-package-versioneer-override: ${{ inputs.versioneer-override }}
+      python-package-build-tag: ${{ inputs.build-tag }}
+
+      skbuild-configure-options: "-DCUDF_BUILD_WHEELS=ON -DDETECT_CONDA_ENV=OFF"
+
+      test-extras:  test
+
+      # Have to manually specify the cupy install location on arm.
+      # Have to also manually install tokenizers==0.10.2, which is the last tokenizers
+      # to have a binary aarch64 wheel available on PyPI
+      # Otherwise, the tokenizers sdist is used, which needs a Rust compiler
+      test-before-arm64: "pip install tokenizers==0.10.2 cupy-cuda11x -f https://pip.cupy.dev/aarch64"
+
+      test-unittest: "pytest -v -n 8 ./python/cudf/cudf/tests"
```

Comment:
Is `-n 8` too aggressive? @galipremsagar says it will crash GPUs depending on the pytest load balancing, and I think we may be seeing that right now with GitHub Actions in PR 12002 (no crosslink to avoid the noise).

Reply:
Were we able to run the pytests with 8 parallel cores in this PR? If yes, could someone link me to the job?

Reply:
Here are the runs @galipremsagar https://github.com/rapidsai/cudf/actions/runs/3476986943/jobs/5812879765

Reply:
Cool. Looks good to me. `-n 8` had been aggressive on v100's, looks like the github action job ran on A100. So 8GB of extra GPU memory is probably helping here. We can leave it at `-n 8` if this workflow is guaranteed to be run on A100's all the time.

Reply:
That's probably a question for @sevagh or @Ethyling. Is this level of parallelism reliable on our CI runners?

Reply:
Discussed this offline. Tests seem fine for now, so we're going to run with the current parallelism and adjust downstream if we start seeing failures.